File: src/main/java/at/hannibal2/skyhanni/utils/jsonobjects/EnigmaSoulsJson.java
package at.hannibal2.skyhanni.utils.jsonobjects;

import at.hannibal2.skyhanni.utils.LorenzVec;
import com.google.gson.annotations.Expose;

import java.util.List;
import java.util.Map;

public class EnigmaSoulsJson {
    @Expose
    public Map<String, List<EnigmaPosition>> areas;

    public static class EnigmaPosition {
        @Expose
        public String name;
        @Expose
        public LorenzVec position;
    }
}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/EnigmaSoulsJson.kt
package at.hannibal2.skyhanni.data.jsonobjects.repo

import com.google.gson.annotations.Expose
import net.minecraft.world.phys.Vec3

data class EnigmaSoulsJson(
    @Expose val areas: Map<String, List<EnigmaPosition>>,
)

data class EnigmaPosition(
    @Expose val name: String,
    @Expose val position: Vec3,
)
